src/Microsoft.DncEng.SecretManager/SecretTypes/PgpKeyPair.cs

```diff
@@ -0,0 +1,31 @@
+using Microsoft.DncEng.CommandLineLib;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.DncEng.SecretManager.SecretTypes
+{
+    [Name("pgp-key-pair")]
+    internal class PgpKeyPair : SecretType<PgpKeyPair.Parameters>
+    {
+        private readonly IConsole _console;
+
+        public class Parameters { }
+    
+        public PgpKeyPair(IConsole console)
+        {
+            _console = console;
+        }
+
+        protected override async Task<SecretData> RotateValue(Parameters parameters, RotationContext context, CancellationToken cancellationToken)
+        {
+            // Adding this to suppress the warning about not having the 'await' operator in an async method
+            await context.GetSecretValue(new SecretReference(context.SecretName));
+
+            throw new HumanInterventionRequiredException($"Pgp key pair secret rotation required. Human intervention required. Please go to https://aka.ms/signalr-java-publishing and follow the instructions at the bottom of the page");
```

Comment:
This page is heavily skewed towards the signalR pgp key. If you're going to provide instructions in a wiki, it should be to a page for generic pgp key generation.

If this particular secret can't be cycled and instead needs to be extended forever, that's a note for the secret itself, not for secret-manager.

Reply:
We talked to Barry about this, extending the key also changes it's private and public key, so it is cycling it, without having to create a new one and doing so isn't dangerous, as long as the key hasn't been compromised. It is something we can consider doing for all future PGP keys

Reply:
If this is the approach for all future secrets of this type, then I'd like to see the code to automatically rotate included in this PR. Let's not add new secret types that are prone to manual mistakes and have so many requisites.

Reply:
At the very least, this still needs to point to a generic document that is not related to signalR.

Reply:
Before we commit to adding PGP key rotation to the Secret Manager, I'd like to try to the other, ESRP route.
I will create a separate page for rotating PGP keys in general

Reply:
That sounds completely reasonable.

Reply:
Would it make sense to hold on this PR until we're sure we need to add this type of secret to our vaults? You mentioned you would be looking into the ESRP task early next year after the release?

Reply:
I think it would, let's go with that